pyccel/codegen/printing/cwrappercode.py

```diff
@@ -56,1026 +50,697 @@ class CWrapperCodePrinter(CCodePrinter):
-                error = PyErr_SetString('PyExc_NotImplementedError',
-                        '"Argument does not have the expected ordering ({})"'.format(collect_var.order))
-                body += [IfSection(PyccelNot(check), [error, Return([Nil()])])]
-
-        body += [IfSection(LiteralTrue(), [Assign(VariableAddress(variable),
-                            self.get_collect_function_call(variable, collect_var))])]
-        body = [If(*body)]
 
-        return body
+        func_name       = 'py_to_nd{}'.format(self._print(variable.dtype))
+        func_name       = self.get_new_name(used_names, func_name)
+        py_variable     = Variable(name = 'py_variable', dtype = PyccelPyObject(), is_pointer = True)
+        c_variable      = variable.clone(name = 'c_variable', is_pointer = True)
+        body            = []
 
-    def _body_management(self, used_names, variable, collect_var, check_type = False):
-        """
-        Responsible for calling functions that take care of body creation
-        Parameters:
-        ----------
-        used_names : list of strings
-            List of variable and function names to avoid name collisions
-        Variable : Variable
-            The optional variable
-        collect_var : variable
-            the pyobject type variable  holder of value
-        check_type : Boolean
-            True if the type is needed
+        # (Valued / Optional) variable check
+        if isinstance(variable, ValuedVariable):
+            body.append(self.generate_valued_variable_body(py_variable, c_variable, variable.value))
 
-        Returns
-        -------
-        body : list
-            A list of statements
-        tmp_variable : Variable
-            temporary variable to hold value default None
-        """
-        tmp_variable = None
-        body         = []
+        body = [Return([PyArray_to_C(py_variable, c_variable,
+                            check_is_needed, self._target_language)])]
 
-        if variable.rank > 0:
-            body = self._body_array(variable, collect_var, check_type)
+        funcDef = FunctionDef(name     = func_name,
+                            arguments  = [py_variable, c_variable],
+                            results    = [Variable(dtype = NativeInteger(), name = 'r', is_temp = True)],
+                            body       = body)
 
-        else:
-            if variable.is_optional:
-                tmp_variable = Variable(dtype=variable.dtype, precision = variable.precision,
-                                        name = self.get_new_name(used_names, variable.name+"_tmp"))
+        return funcDef
 
-            body = [self._body_scalar(variable, collect_var, check_type, tmp_variable)]
+    def generate_tuple_converter_function(self, used_names, variable):
+        pass
+        #TODO
 
-        return body, tmp_variable
+    def generate_list_converter_function(self, used_names, variable):
+        pass
+        #TODO
```

Comment:
Will this convert to or from list? We don't handle lists so you don't need to convert to a list. We just want a permissive version of tuple collection

Reply:
from pylist  to c array

Reply:
anyway  im gonna remove it later. it will be handled same as tuple